api/v1_stats.inc

```diff
@@ -11,10 +11,22 @@ function api_v1_stats_site($method, $data, $query_params)
     $res = DPDatabase::query("SELECT COUNT(*) FROM users");
     list($registered_users) = mysqli_fetch_row($res);
 
-    return [
+    $stats = [
         "server_time" => date(DATE_ATOM),
         "registered_users" => $registered_users,
     ];
+
+    foreach ( array(1,7,30) as $days_back )
+    {
+        $res = DPDatabase::query("
```

Comment:
As part of preventing SQL injections, we've been using `sprintf()` to insert string and integer variables, like `$days_back`, into SQL. Even here where it's not user input we'd like to use it because some developer down the road may make it user-selectable and then we have a vulnerability. eg:

```php
    $sql = sprintf("
        SELECT COUNT(*)
        FROM users
        WHERE t_last_activity > UNIX_TIMESTAMP() - %d * 24*60*60
    ", $days_back);
    $res = DPDatabase::query($sql);
```

Reply:
ACK. Done.